core/src/main/scala/IndexedItemized.scala

```diff
@@ -21,9 +21,11 @@ trait IndexedItemized extends Itemized {
 }
 
 trait ItemizedIndex[T <: IndexedItemized] {
-  def toIndex(c: T): T#Index
+  def toIndex(c: T): T#Index = c.index
 
-  def fromIndex(v: T#Index): Option[T]
+  def fromIndex(v: T#Index): Option[T] = indexMap.get(v)
+
+  val indexMap: Map[T#Index, T]
```

Comment:
Sure, that would work perfectly for my case!
My only "concern" is that fromIndex and toIndex seem a bit redundant now.